packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -626,6 +603,62 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		}
 	}
 
+	private createRevertible(
+		revision: RevisionTag,
+		kind: CommitKind,
+		checkout: TreeCheckout,
+		onRevertibleDisposed: (revertible: Revertible) => void,
+	): RevertibleAlpha {
+		const commitBranches = checkout.revertibleCommitBranches;
+
+		const revertible: RevertibleAlpha = {
+			get status(): RevertibleStatus {
+				const revertibleCommit = commitBranches.get(revision);
+				return revertibleCommit === undefined
+					? RevertibleStatus.Disposed
+					: RevertibleStatus.Valid;
+			},
+			revert: (release: boolean = true) => {
+				if (revertible.status === RevertibleStatus.Disposed) {
+					throw new UsageError("Unable to revert a revertible that has been disposed.");
+				}
+
+				const revertMetrics = checkout.revertRevertible(revision, kind);
+				checkout.logger?.sendTelemetryEvent({
+					eventName: TreeCheckout.revertTelemetryEventName,
+					...revertMetrics,
+				});
+
+				if (release) {
+					revertible.dispose();
+				}
+			},
+			clone: (forkedBranch?: TreeBranch) => {
+				if (forkedBranch === undefined) {
+					return this.createRevertible(revision, kind, checkout, onRevertibleDisposed);
+				}
+
+				const forkedCheckout = getCheckout(forkedBranch);
+				const revertibleBranch = this.revertibleCommitBranches.get(revision);
+				assert(revertibleBranch !== undefined, "SharedTreeBranch for revertible not found.");
+				forkedCheckout.revertibleCommitBranches.set(revision, revertibleBranch.fork());
```

Comment:
Please add a TODO above this line to remind us to optimize this in the future. A fork only needs to be created once _per revision_, not once _per revision per checkout_. Add a work item to ADO and link the work item as well.